### PR TITLE
JIT: update block weight for uncond to cond flow opt

### PR DESCRIPTION
**Pull request by AndyAyersMS**
This optimization duplicates code and flow in a BBJ_COND successor into one of its preds; as a result the weight of the successor should decrease.

Fixes some issues seen with odd perf scores in the ML/CSE experiment.

Contributes to #93020

[Diffs](https://dev.azure.com/dnceng-public/public/_build/results?buildId=580263&view=ms.vss-build-web.run-extensions-tab)

### CODE REVIEW

**Comment by AndyAyersMS** [comment]
FYI @dotnet/jit-contrib 

A few large local regressions. The ones I looked at were all additional cloning in loops with type tests.

**Review by amanasifkhalid — APPROVED** [review]
LGTM, thanks for the fix! I anticipate those additional `fgGetPredForBlock` calls will go away soon, once I replace `bbTrueTarget` and `bbFalseTarget` with flow edges.

File: src/coreclr/jit/fgopt.cpp [hunk 1]
```diff
@@ -2471,24 +2471,39 @@ bool Compiler::fgOptimizeUncondBranchToSimpleCond(BasicBlock* block, BasicBlock*
     // add an unconditional block after this block to jump to the target block's fallthrough block
     //
     assert(!target->IsLast());
-    BasicBlock* next = fgNewBBafter(BBJ_ALWAYS, block, true, target->GetFalseTarget());
+    BasicBlock* const next = fgNewBBafter(BBJ_ALWAYS, block, true, target->GetFalseTarget());
```
[the remark this answers is not in the document]
Comment (AndyAyersMS) [reply src/coreclr/jit/fgopt.cpp:2474]: I had removed the fixup block originally, but put it back to reduce diffs.

**Comment by AndyAyersMS** [comment]
Hmm, rather bigger diffs than I was expecting. 

I will need to dig in and see if this is all attributable to more cloning, and whether it is time to at least build some kind of vague heuristic.

**Comment by AndyAyersMS** [comment]
Looks like regressions are indeed from more cloning.

**Comment by AndyAyersMS** [comment]
In particular type test cloning is driven by the likelihood of the type test succeeding, and with this profile update we now see more tests that appear successful.

**Comment by AndyAyersMS** [comment]
@amanasifkhalid can you take another look? I removed the Next block and just wire up the flow directly.

TP diffs good, PerfScore diffs good. Code size increases, but mainly from libraries tests. Code size impact is all from more or fewer clones, all the ones I saw were from the "clone for type test" heuristic which relies on profile data.

**Review by amanasifkhalid — APPROVED** [review]
LGTM, thanks for getting rid of some of the "no fallthrough" cruft.

**Comment by AndyAyersMS** [comment]
Failure is a timeout spmi replay for linux arm32.

**Comment by EgorBo** [comment]
Improvements on arm64:

- https://github.com/dotnet/perf-autofiling-issues/issues/30175

**Comment by EgorBo** [comment]
Improvements:

- https://github.com/dotnet/perf-autofiling-issues/issues/30633

**Pull request merged**